test/e2e/specs/site-editor/new-templates-list.spec.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Templates', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await Promise.all( [
+			requestUtils.activateTheme( 'emptytheme' ),
+			requestUtils.activatePlugin( 'gutenberg-test-dataviews' ),
+		] );
+	} );
+	test.afterAll( async ( { requestUtils } ) => {
+		await Promise.all( [
+			requestUtils.activateTheme( 'twentytwentyone' ),
+			requestUtils.deactivatePlugin( 'gutenberg-test-dataviews' ),
+		] );
+	} );
+	test( 'Sorting', async ( { admin, page } ) => {
+		await admin.visitSiteEditor( { path: '/wp_template/all' } );
+		// Descending by title.
+		await page.getByRole( 'button', { name: 'Template' } ).click();
+		await page.getByRole( 'menuitem', { name: 'Sort descending' } ).click();
```

Comment:
This is just nitpicking but we can also scope it under a menu to make this slightly clearer that we're clicking inside a dropdown:

```js
await page
  .getByRole( 'menu', { name: 'Template' } )
  .getByRole( 'menuitem', { name: 'Sort descending' } )
  .click();
```